infra/base-images/base-runner/test_all.py

```diff
@@ -162,16 +164,94 @@ def test_all_outside_out(allowed_broken_targets_percentage):
     return test_all(out, allowed_broken_targets_percentage)
 
 
+def find_centipede_auxiliary_directory(directory):
+  """Finds the directory that contains sanitized targets for Centipede."""
+  # The sanitized target binaries are always in a child directory named
+  # '__centipede_${SANITIZER}'.
+  sanitized_binary_dir_path = os.path.join(
+      directory, f'__centipede_{os.getenv("SANITIZER")}')
+  if os.path.isdir(sanitized_binary_dir_path):
+    return sanitized_binary_dir_path
+
+  # This should never happen, returns an empty string to indicate an error.
+  print(
+      'ERROR: Unable to identify Centipede\'s sanitized target directory from',
+      os.listdir(directory))
+  return ''
+
+
+def centipede_needs_auxiliaries():
+  """Checks if auxiliaries are needed for Centipede."""
+  # Centipede always requires unsanitized binaries as the main fuzz targets,
+  # and separate sanitized binaries as auxiliaries.
+  # 1. Building sanitized binaries with helper.py (i.e., local or GitHub CI):
+  # Unsanitized ones will be built automatically into the same docker container.
+  # Script bad_build_check tests both
+  # a) if main fuzz targets can run with the auxiliaries, and
+  # b) if the auxiliaries are built with the correct sanitizers.
+  # 2. In Trial build and production build:
+  # Two kinds of binaries will be in separated buckets / docker containers.
+  # Script bad_build_check tests either
+  # a) if the unsanitized binaries can run without the sanitized ones, or
+  # b) if the sanitized binaries are built with the correct sanitizers.
+  return (os.getenv('FUZZING_ENGINE') == 'centipede' and
+          os.getenv('SANITIZER') != 'none' and os.getenv('HELPER') == 'True')
```

Comment:
Done. I guess this is because we won't set `HELPER` if we are not building with `helper.py`, but is it safer to compare the value? (e.g. just in case some future commit sets it to something else.)

Reply:
Nevermind, just compare it to True so we'll handle "False" properly.